data/repository/src/commonMain/kotlin/social/androiddev/common/repository/AuthenticationRepositoryImpl.kt

```diff
@@ -32,7 +32,7 @@ internal class AuthenticationRepositoryImpl(
         redirectUris: String,
         scopes: String,
         website: String?
-    ): NewAppOAuthToken? {
+    ): NewAppOAuthToken? = withContext(ioCoroutineContext) {
```

Comment:
I've noticed we're not making this `main` safe, so added withContext. There are no suspend points in those methods, however, so it'll fallback to sync anyway

Reply:
My 50c here, `mastodonApi.createApplication` is already a `suspend` function, switching contexts should be performed on the callee site, not caller. E.g. Ktor as async by default.

Reply:
Aw yeah, I do agree with that - I'll create a follow up PR to inject context to ApiKtor client and switch context there